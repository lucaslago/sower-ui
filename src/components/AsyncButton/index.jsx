import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import CircularProgress from 'material-ui/CircularProgress';

const AsyncButton = props => {
  if(props.makingRequest) {
    return (<CircularProgress />);
  }
  return (<RaisedButton label={ props.label } type={ props.type }/>);
};

export default AsyncButton;
