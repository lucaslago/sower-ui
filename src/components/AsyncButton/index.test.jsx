import AsyncButton from './index';
import RaisedButton from 'material-ui/RaisedButton';
import CircularProgress from 'material-ui/CircularProgress';

describe('<AsyncButton />', () => {
  const label = 'login';
  const type = 'text';
  let makingRequest = false;

  it('renders correctly', () => {
    const wrapper = shallow(<AsyncButton type={type} label={label} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders <RaisedButton /> when makingRequest is false', () => {
    const wrapper = shallow(<AsyncButton label={label} makingRequest={makingRequest} />);
    expect(wrapper.find(RaisedButton).length).toEqual(1);
    expect(wrapper.find(CircularProgress).length).toEqual(0);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders <CircularProgress /> when makingRequest is true', () => {
    const wrapper = shallow(
      <AsyncButton label="login" makingRequest={ true } />
    );
    expect(wrapper.find(RaisedButton).length).toEqual(0);
    expect(wrapper.find(CircularProgress).length).toEqual(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
