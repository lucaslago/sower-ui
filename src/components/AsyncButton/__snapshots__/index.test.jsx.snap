exports[`<AsyncButton /> renders <CircularProgress /> when makingRequest is true 1`] = `
<CircularProgress
  max={100}
  min={0}
  mode="indeterminate"
  size={40}
  thickness={3.5}
  value={0} />
`;

exports[`<AsyncButton /> renders <RaisedButton /> when makingRequest is false 1`] = `
<RaisedButton
  disabled={false}
  fullWidth={false}
  label="login"
  labelPosition="after"
  primary={false}
  secondary={false} />
`;

exports[`<AsyncButton /> renders correctly 1`] = `
<RaisedButton
  disabled={false}
  fullWidth={false}
  label="login"
  labelPosition="after"
  primary={false}
  secondary={false}
  type="text" />
`;
